Sources/FirebaseAuth/FirebaseAuthError.swift

```diff
@@ -1,117 +1,150 @@
 // SPDX-License-Identifier: BSD-3-Clause
 
-import Foundation
+@_exported
+import firebase
+@_spi(FirebaseInternal)
+import FirebaseCore
 
-public enum AuthErrorCode: Int {
-  case invalidCustomToken = 2
-  case customTokenMismatch = 3
-  case invalidCredential = 4
-  case userDisabled = 5
-  case accountExistsWithDifferentCredential = 6
-  case operationNotAllowed = 7
-  case emailAlreadyInUse = 8
-  case requiresRecentLogin = 9
-  case credentialAlreadyInUse = 10
-  case invalidEmail = 11
-  case wrongPassword = 12
-  case tooManyRequests = 13
-  case userNotFound = 14
-  case providerAlreadyLinked = 15
-  case noSuchProvider = 16
-  case invalidUserToken = 17
-  case userTokenExpired = 18
-  case networkError = 19
-  case invalidAPIKey = 20
-  case appNotAuthorized = 21
-  case userMismatch = 22
-  case weakPassword = 23
-  case noSignedInUser = 24
-  case apiNotAvailable = 25
-  case expiredActionCode = 26
-  case invalidActionCode = 27
-  case invalidMessagePayload = 28
-  case invalidPhoneNumber = 29
-  case missingPhoneNumber = 30
-  case invalidRecipientEmail = 31
-  case invalidSender = 32
-  case invalidVerificationCode = 33
-  case invalidVerificationID = 34
-  case missingVerificationCode = 35
-  case missingVerificationID = 36
-  case missingEmail = 37
-  case missingPassword = 38
-  case quotaExceeded = 39
-  case retryPhoneAuth = 40
-  case sessionExpired = 41
-  case appNotVerified = 42
-  case appVerificationUserInteractionFailure = 43
-  case captchaCheckFailed = 44
-  case invalidAppCredential = 45
-  case missingAppCredential = 46
-  case invalidClientID = 47
-  case invalidContinueURI = 48
-  case missingContinueURI = 49
-  case keychainError = 50
-  case missingAppToken = 51
-  case missingIosBundleID = 52
-  case notificationNotForwarded = 53
-  case unauthorizedDomain = 54
-  case webContextAlreadyPresented = 55
-  case webContextCancelled = 56
-  case dynamicLinkNotActivated = 57
-  case cancelled = 58
-  case invalidProviderID = 59
-  case webInternalError = 60
+public struct AuthErrorCode: RawRepresentable, Error {
+  public typealias RawValue = Int
+
+  public let rawValue: Int
+  public let localizedDescription: String
+
+  public init(rawValue: Int) {
```

Comment:
Might be nice to add a doc comment to explain why this is `public` - we don't want others to be able to create the error I think?

Reply:
It needs to be public as part of the `RawRepresentable` conformance.